components/WiFiComponent/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+idf_component_register(SRCS "WiFiConfig_.c"
+                    INCLUDE_DIRS "include"
+                    REQUIRES  esp_wifi esp_eth protocol_examples_common
+                    esp_http_client nvs_flash
+                    main)
```

Comment:
main component should not add to any component

Reply:
Actually main component can  be 😀 , in the STM , NXP  and .. library 's , this  is main.h  ,and is safe , don't have any problem